packages/zero-client/src/client/replicache-types.ts

```diff
@@ -34,7 +35,8 @@ export interface ReadTransaction extends ReplicacheReadTransaction {
   readonly env?: Env | undefined;
 }
 
-export interface WriteTransaction extends ReplicacheWriteTransaction {
+export interface WriteTransaction<T extends ZeroTxData = ZeroTxData>
```

Comment:
I thought we did this... I don't yet understand why we have to change the Replicache interface.

Reply:
When Replicache starts a rebase it'll ask Zero (via zeroOption) to create a fork of the IVM sources at the given head. This fork will be added to `WriteTransaction` so `Zero` can use it in it's mutators.

Previously I assumed there were a finite (or at least small) set of heads and that Zero could look them up by name rather than having to pass the fork along in the Replicache transaction. It turns out there isn't a small set of heads being rebased onto. Looking up by name also opens us up to concurrency issues.

Reply:
Let me put up the PR that actually uses this argument.

Reply:
I think I see the big picture better now. I still wish we could keep the Replicache public interfaces as they are.

I have a few tricks up my sleeves that would allow this to work without changing the interface. I think it is OK to add public properties (using symbols) to the implementation without changing the interface for example. Other options include using weakmaps as sidetables.

Reply:
hmm, maybe I can add it to the args going to `MutatorImpl` and all `Zero` mutators will require this argument.